### ISSUE TITLE
Website selection drop-down: poor usability if more than 15 websites

### ISSUE THREAD

**Issue by ma77ie** [matomo-org/matomo#15546]
The website selection drop-down in the top left of the Dashboard allows you to select which of the websites tracked in Matomo you wish to look at. If you have more than 15 websites on Matomo it appears to show the first 15 added in alphabetical order and subsequently added websites never appear. Because the list is A to Z it looks like the complete list of available list, so I often get complaints that users can't see their website because its not being shown by default. I appreciate you can search for a website (but sometimes the user doesn't know the actual name of the website) or click on All Websites (but they think they can see all websites already because of the A to Z list they can see). It would be far better if all websites where listed in the selection drop-down (could have a scroll bar if its a long list).

**Comment by tsteur** [comment]
FYI in case it's helpful, there is a setting to show more sites (but it probably wouldn't have scrollbars) https://matomo.org/faq/how-to/faq_23986/

We would likely not show all websites since some have eg Matomo running with 40K websites etc. and it would make things very slow even just to load that data. We could potentially show something like `9 of 40 sites` or something else though.

**Comment by ma77ie** [comment]
Thanks very much for that, Thomas

That solves my problem.

Regards,
Matt

On Monday, February 10, 2020 19:01 GMT, Thomas Steur <notifications@github.com> wrote:
 
FYI in case it's helpful, there is a setting to show more sites (but it probably wouldn't have scrollbars) https://matomo.org/faq/how-to/faq_23986/
We would likely not show all websites since some have eg Matomo running with 40K websites etc. and it would make things very slow even just to load that data. We could potentially show something like 9 of 40 sites or something else though.

**Comment by skylarmt** [comment]
I'd love to see the website drop-down be customizable. I have a few dozen websites, but I often find myself checking on just a few of them. It would be nice if I could re-order the list so those are at the top.

**Comment by heurteph-ei** [comment]
@skylarmt 
Workaround: As the list is A to Z, you can prefix the site name with a number that would sort the list displayed in the dropdown...

**Comment by Stan-vw** [comment]
Unfortunately I ran into some trouble when trying to replicate this (couldn't add 15+ websites), could you share how it currently looks when you have many websites?
Just to check, we're talking about the dropdown element in the attached screenshot right?

Seems like a decent idea to have it have a scrollbar if many websites are added. Thoughts by you @Javi-Ormaechea ? 
![image](https://user-images.githubusercontent.com/52564377/228079454-5f318927-3dd7-4eed-8aca-7000fad6672a.png)

**Comment by Javi-Ormaechea** [comment]
@Stan-vw I agree with making it scrollable, and keeping the search field to get quick access to a site that might be found at the end of that list. Another thing we could do, is to provide a list of the last 4 viewed sites so that users can quickly get back to those if needed, something like this:

<img width="796" alt="option-02" src="https://user-images.githubusercontent.com/109315616/228113642-7acb930a-bfe4-44dd-9960-2e1341d1018f.png">

<img width="801" alt="option-01" src="https://user-images.githubusercontent.com/109315616/228113635-e32dc460-5ff1-4bae-9a89-6be053f0cbfa.png">

**Comment by pbb72** [comment]
@heurteph-ei That wouldn't help, as the list still only shows the 15 oldest sites. If you have more than 15 sites, there is nothing you can do to get the newer ones on the list, apart from deleting older sites.

**Comment by pbb72** [comment]
Note that the site selector is also used on the Manage Users page (located under Administration), but this list is missing the "All websites" link, making it even worse usability.

**Comment by Chardonneaur** [comment]
Asked by a partner today.

**Comment by LauraAndro** [comment]
Another customer asked the same.
